firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java

```diff
@@ -130,6 +132,27 @@ public void persistNonFatalEvent(Throwable event, Thread thread, long timestamp)
     persistEvent(event, thread, EVENT_TYPE_LOGGED, timestamp, false);
   }
 
+  public void finalizeNativeEvent(
+      @NonNull List<NativeSessionFile> nativeSessionFiles, String sessionId) {
+    try {
+      FilesPayload.Builder filesPayloadBuilder = FilesPayload.builder();
+      ArrayList<FilesPayload.File> nativeFiles = new ArrayList<>();
+      for (NativeSessionFile nativeSessionFile : nativeSessionFiles) {
+        FilesPayload.File filePayload = nativeSessionFile.asFilePayload();
+        if (filePayload != null) {
+          nativeFiles.add(filePayload);
+        }
+      }
+
+      filesPayloadBuilder.setFiles(ImmutableList.from(nativeFiles));
+
+      reportPersistence.finalizeNativeEvent(
```

Comment:
Same as above, can we call this `finalizeSessionWithNativeEvent(sessionID, reportData, filesPayload)`

Reply:
sure